components/partials/modals/referral-onboarding.vue

```diff
@@ -0,0 +1,182 @@
+            class="inline-flex items-center justify-center"
+            light
+            :disabled="metamaskStatus.isLoading()"
+            @click="connectLedger"
+          >
+            <v-icon-ledger class="w-6 h-6 mr-3" />
+            <span class="font-bold tracking-widest">
+              {{ $t('connect.ledger') }}
+            </span>
+          </v-button>
+        </div>
+      </div>
+
+      <div v-else>
+        <div class="flex items-center my-6">
+          <div
+            class="bg-gray-200 mr-3 rounded-full h-10 w-10 flex items-center justify-center"
+          >
+            <v-icon-metamask
+              v-if="connectedWalletType === Wallet.Metamask"
+              class="w-6 h-6"
+            />
+            <v-icon-ledger
+              v-else-if="connectedWalletType === Wallet.Ledger"
+              class="w-6 h-6 text-gray-700"
+            />
+          </div>
+          <span class="text-xs tracking-wide">{{ injectiveAddress }}</span>
+        </div>
+
+        <v-button primary xl class="w-full" :disabled="invalid">
+          <span class="font-bold">
+            {{ $t('referralModal.confirm') }}
+          </span>
+        </v-button>
+      </div>
+    </ValidationObserver>
+  </v-modal>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { ValidationObserver, ValidationProvider } from 'vee-validate'
+import { Status, StatusType } from '@injectivelabs/utils'
+import { Wallet } from '@injectivelabs/web3-strategy'
+import { AccountAddress } from '@injectivelabs/ts-types'
+import { Modal, WalletConnectStatus } from '~/types'
+
+export default Vue.extend({
+  components: {
+    ValidationProvider,
+    ValidationObserver
+  },
+
+  data() {
+    return {
+      Wallet,
+      metamaskStatus: new Status(StatusType.Idle),
+      form: {
+        code: ''
+      }
+    }
+  },
+
+  computed: {
+    isModalOpen(): boolean {
+      return this.$accessor.modal.modals[Modal.ReferralOnboarding]
+    },
+
+    isUserWalletConnected(): boolean {
+      return this.$accessor.wallet.isUserWalletConnected
+    },
+
+    injectiveAddress(): AccountAddress {
+      return this.$accessor.wallet.injectiveAddress
+    },
+
+    connectedWalletType(): Wallet {
+      return this.$accessor.wallet.wallet
+    },
+
+    $form(): InstanceType<typeof ValidationObserver> {
+      return this.$refs.form as InstanceType<typeof ValidationObserver>
+    }
+  },
+
+  mounted() {
+    const { code } = this.$route.query
+    if (code) {
+      this.form.code = code as string
+      this.$nextTick(() => {
+        this.$form.validate()
+      })
+    }
+  },
+
+  methods: {
+    closeModal() {
+      this.$accessor.modal.closeModal(Modal.ReferralOnboarding)
+      this.$accessor.modal.setPersistModal(undefined)
+    },
+
+    handleConfirm() {
+      if (this.$form) {
+        this.$form.reset()
+      }
+
+      this.closeModal()
+    },
+
+    connectLedger() {
+      this.$accessor.modal.setPersistModal(Modal.ReferralOnboarding)
```

Comment:
Can you explain to me why are have we introduced this?

Reply:
On selecting the Ledger login option, we will close the referral onboarding modal and expand the ledger login modal (select address, ...). We persist the modal option in our store so the UX will open the referral modal again upon closing/logging in from the ledger login modal.

Reply:
> On selecting the Ledger login option, we will close the referral onboarding modal and expand the ledger login modal (select address, ...). We persist the modal option in our store so the UX will open the referral modal again upon closing/logging in from the ledger login modal.

I don't think it a good idea to introduce a new codebase just for handling an edge case, instead, we should emit an event when the Ledger modal is closed (or w/e the condition is) and open this modal. If this was a standard use-case where every relayer would for sure use this feature then its fine to introduce this but since it'll be optional, set with a flag, lets try to use existing features to handle the edge cases.